thanos/charts/templates/_helpers.tpl

```diff
@@ -36,4 +36,18 @@ kube-monitoring-prometheus
 {{- else -}}
 object-storage-configs.yaml
 {{- end }}
-{{- end }}
\ No newline at end of file
+{{- end }}
+
+{{/*
+Renders a single `annotations:` block by merging two maps.
+`overrideAnnotations` overrides `baseAnnotations` on key conflicts. Emits nothing if the merged map is empty.
+*/}}
+{{- define "thanos.annotations" -}}
+{{- $baseAnnotations := .a | default dict -}}
+{{- $overrideAnnotations := .b | default dict -}}
+{{- $mergedAnnotations := merge (dict) $baseAnnotations $overrideAnnotations -}}
+{{- if $mergedAnnotations -}}
+annotations:
+{{- toYaml $mergedAnnotations | nindent 2 }}
+{{- end -}}
+{{- end }}
```

Comment:
What do you think about this basic logic to combine the annotations?
```suggestion
{{- define "thanos.annotations" -}}
{{- if or .a .b }}
annotations:
{{- with .a  }}
{{- toYaml . | nindent 2 }}
{{- end }}
{{- with .b }}
{{- toYaml . | nindent 2 }}
{{- end }}
{{- end }}
{{- end }}
```

Reply:
Additionally: I would like to get rid of `a` and `b` entirely. Better readable names > shortness.